include/xtensor-python/xtensor_type_caster_base.hpp

```diff
@@ -23,6 +23,97 @@ namespace pybind11
 {
     namespace detail
     {
+        template <typename T, xt::layout_type L>
+        struct xtensor_get_buffer
+        {
+            static auto get(handle src)
+            {
+                return array_t<T, array::c_style | array::forcecast>::ensure(src);
+            }
+        };
+
+        template <typename T>
+        struct xtensor_get_buffer<T, xt::layout_type::column_major>
+        {
+            static auto get(handle src)
+            {
+                return array_t<T, array::f_style | array::forcecast>::ensure(src);
+            }
+        };
+
+        template <class T>
+        struct xtensor_check_buffer
+        {
+        };
+
+        template <class T, xt::layout_type L>
+        struct xtensor_check_buffer<xt::xarray<T, L>>
+        {
+            static auto get(handle src)
+            {
+                return xtensor_get_buffer<T, L>::get(src);
+            }
+        };
```

Comment:
Nitpicking: in line with the previous renaming proposal:
`xtensor_chek_buffer` => `pybind_array_getter`
`get` => `run`